### DISCUSSION TITLE
Alliance highlighting

### DISCUSSION THREAD

**Discussion by Skilly146** [Tsangares/stoned_ape_tools#12] [Feature Requests]
## What

Add a method for a user to show the extension who is in their alliance as well as other alliances. Then highlight all the stars of each alliance in the same color, similar to the warpgate effect

## How

Since the extension can already read messages in order to use the api sharing feature I assume it can also read them at all times. That means the easiest way to implement this would probably just be a specially formatted message, something like this

> Title Bar -> "Highlight Color/other highlighting options. comments with # to add alliance name for organization"
>
> main body -> "For each player in an alliance use the in game player linking with '[[player id]] name' format to specify players to highlight with above method"

It might be better to just do highlighting options within the main body it was just easier to separate it that way for now

## Similar Existing Tools

qwerty wrote a user script that does something similar a while ago for an old version of the game [here](https://forum.ironhelmet.com/t/map-tweaks-alliances-hide-personal-info/3788). I doubt any of it is still useful since I'm pretty sure that was before the server rewrite but there might be something

**Comment by anicolao** [comment]
This possibly overlaps with my suggestion that all colours of all players be configurable.
